### ISSUE TITLE
Question: can lifecycled process requests for other instances?

### ISSUE THREAD

**Issue by ketzacoatl** [buildkite/lifecycled#80]
One use case for lifecycled is that you run it on all EC2 instances, and when an instance terminates, you run the action on the instance that is terminating.

Another use case is to run those actions on a single/central management instance, rather than the instance that is terminating. There might be a variety of reasons for this, such as reliability, management access or permissions constraints in the env.

With a review of the go code, it seems lifecycled is restricting actions to the instance it is running on, is that true?

**Comment by qrilka** [comment]
#79 is somewhat related to this issue

**Comment by itsdalmo** [comment]
> With a review of the go code, it seems lifecycled is restricting actions to the instance it is running on, is that true?

Yep, each instance will only act on termination events that are targeting its own instance ID. The difficulty with running it centrally is that you'd need another way to initiate the shutdown script on the host that is being terminated, which I believe would increase complexity a lot 🤔

**Comment by ketzacoatl** [comment]
> The difficulty with running it centrally is that you'd need another way to initiate the shutdown script on the host that is being terminated, which I believe would increase complexity a lot 🤔

For some use cases, the actions taken _are_ with a central system and can be run from a single instance responsible for running a tool like lifecycled.

**Comment by ketzacoatl** [comment]
@itsdalmo / @lox, would making the filter optional be enough to make this a reality?
